src/lucky_record/where.cr

```diff
@@ -1,19 +1,53 @@
 module LuckyRecord::Where
   abstract class SqlClause
-    getter :column, :value
+    getter :column
 
-    def initialize(@column : Symbol | String, @value : String | Array(String) | Array(Int32))
+    def initialize(@column : Symbol | String)
     end
 
     abstract def operator : String
     abstract def negated : SqlClause
 
+    def prepare
+      "#{column} #{operator}"
+    end
+  end
+
+  abstract class ComparativeSqlClause < SqlClause
```

Comment:
I think I understand why you introduced this class. Inheriting from `SqlClause` and overriding the constructor to only accept the `@column` will throw an error unless you make `@value` nilable, which would break the class.

Your solution does work well to separate the two concepts, but I think clauses that reference `NULL` are a special case and should inherit from `SqlClause` instead of the other way around.

You can achieve this without changing `SqlClause` by introducing a `NullSqlClause` class and removing `@value` from the constructor by setting it to `"NULL":

```crystal
abstract class SqlClause
  getter :column, :value

  def initialize(@column : Symbol | String, @value : String | Array(String) | Array(Int32))
  end

  abstract def operator : String
  abstract def negated : SqlClause

  def prepare(prepared_statement_placeholder : String)
    "#{column} #{operator} #{prepared_statement_placeholder}"
  end
end

abstract class NullSqlClause < SqlClause
  @value = "NULL"

  def initialize(@column : Symbol | String)
  end
end
```
The `Null` and `NotNull` classes can then use the default `prepare` method to generate their sql so you can set their operators to "IS" and "IS NOT":

```crystal
class Null < NullSqlClause
  def operator
    "IS"
  end
  ...
end
```
I believe this also means you can use the normal `prepared_statement_values` and `joined_wheres_queries` methods.

Reply:
Hmm yeah I like this idea better. It's likely to be the only clause of its sort.

Reply:
One thing: I'm not sure how you can get `NullSqlClause` to use the normal `prepared_statement_values` as that would result with `users.nickname IS NULL $1`.

Reply:
If you change the operator to "IS" it should come out like `users.nickname IS $1` and work  as expected, right?

Reply:
Ah yes. Sorry!

Reply:
Sorry to keep asking questions about this, but won't passing in `NULL` as a dollar-sign param. result in it getting escaped? And just treated like the string `"NULL"`? And we can't override the ivar in the child class to set it to a `nil` literal.

Reply:
It's no problem, I missed the details and oversimplified it. I think you can do with another prepare method then :)
```
def prepare
   "#{column} #{operator} #{@value}"
end
```